datalad/support/annexrepo.py

```diff
@@ -1153,6 +1153,93 @@ def _call_annex_records(self, args, files=None, jobs=None,
 
         return return_objects
 
+    def _call_annex_records_items_(self,
+                                   args,
+                                   files=None,
+                                   jobs=None,
+                                   git_options=None,
+                                   stdin=None,
+                                   merge_annex_branches=True,
+                                   progress=False,
+                                   **kwargs):
+        """Yielding git-annex command execution with JSON result processing
+
+        `_call_annex()` is used for git-annex command execution, using
+        GeneratorAnnexJsonProtocol. This means _call_annex() will yield
+        results as soon as they are available.
+
+        For a description of the parameters and raised exceptions, please
+        refer to _call_annex_records().
+
+        Returns
+        -------
+        Generator(something)
+        list(dict)
+          List of parsed result records.
+        """
+        protocol_class = GeneratorAnnexJsonProtocol
+
+        args = args[:] + ['--json', '--json-error-messages']
+        if progress:
+            args += ['--json-progress']
+
+        json_objects_received = False
+        try:
+            for json_object in self._call_annex(
+                                      args,
+                                      files=files,
+                                      jobs=jobs,
+                                      protocol=protocol_class,
+                                      git_options=git_options,
+                                      stdin=stdin,
+                                      merge_annex_branches=merge_annex_branches,
+                                      **kwargs):
+                if len(json_object) == 1 and json_object.get('info', None):
+                    lgr.info(json_object['info'])
+                else:
+                    json_objects_received = True
+                    yield json_object
+
+        except CommandError as e:
+            # Note: Workaround for not existing files as long as annex doesn't
+            # report it within JSON response:
+            # see http://git-annex.branchable.com/bugs/copy_does_not_reflect_some_failed_copies_in_--json_output/
+            not_existing = [
+                # cut the file path from the middle, no useful delimiter
+                # need to deal with spaces too!
+                line[11:-10] for line in e.stderr.splitlines()
+                if line.startswith('git-annex:') and
+                line.endswith(' not found')
+            ]
+
+            yield from (
+                {
+                    "command": args[0],
+                    "file": f,
+                    "note": "not found",
+                    "success": False,
+                }
+                for f in not_existing
+            )
+
```

Comment:
the entire body of except handling above here is duplicated now with the one in `_call_annex_records` .  worth centralizing it into a generator so here it would just become

```suggestion
            yield from _handle_not_found(e.stderr)
```
or alike, and in  _call_annex_records just make it into the list

Reply:
I am gonna take care of that, following a merge of #6550, since resolving conflict that way will be easier (and either way likely easiet for me ATM)